### PR TITLE
Ignore `_opam` in git and check-typo.

### PR DESCRIPTION
**Pull request by arbipher**
This PR is for issue #10648.

### CODE REVIEW

**Review by gasche — APPROVED** [review]
Looks good to me!

(Re. @kit-ty-kate's remark on redundancy: the testing seems to indiate that `typo.prune` on directories is more efficient; maybe we should change this, but in the meantime this is simple and works well.)

File: .gitattributes [hunk 1]
```diff
@@ -70,6 +70,8 @@ tools/mantis2gh_stripped.csv typo.missing-header
 # ignore auto-generated .depend files
 .depend                  typo.prune
 /.depend.menhir          typo.prune
+# ignore local opam switch
+/_opam                   typo.prune
```
Comment (dra27) [inline .gitattributes]: This becomes unnecessary if/when #10656 is merged

**Review by dra27 — COMMENTED** [review]
Likewise LGTM, thanks! We can either merge this as-is and update #10656, or vice versa.

File: .gitattributes (hunk 1 above)
Comment (dra27) [reply .gitattributes]: (which it now is!)

Comment (arbipher) [reply .gitattributes]: I update my commit (and I know the difference between moving a branch and rebasing it for the first time)

**Comment by dra27** [comment]
Thanks!

**Pull request merged**